azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -934,6 +941,14 @@ func flattenAzureRmVirtualMachineScaleSetNetworkProfile(profile *compute.Virtual
 					config["subnet_id"] = *properties.Subnet.ID
 				}
 
+				if properties.ApplicationGatewayBackendAddressPools != nil {
+					addressPools := make([]interface{}, 0, len(*properties.ApplicationGatewayBackendAddressPools))
+					for _, pool := range *properties.ApplicationGatewayBackendAddressPools {
+						addressPools = append(addressPools, *pool.ID)
+					}
+					config["application_gateway_backend_address_pool_ids"] = schema.NewSet(schema.HashString, addressPools)
```

Comment:
so we probably want to set a Set regardless here (e.g. to an empty list if it's not specified) - as such could we make this:

```
addressPools := make([]interface{}, 0)
if properties.ApplicationGatewayBackendAddressPools != nil {
  for _, pool := range *properties.ApplicationGatewayBackendAddressPools {
    addressPools = append(addressPools, *pool.ID)
  }
}
config["application_gateway_backend_address_pool_ids"] = schema.NewSet(schema.HashString, addressPools)
```